xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiWikiReference.java

```diff
@@ -38,12 +39,31 @@ public class XWikiWikiReference extends WikiReference
 
     private boolean freeStanding;
 
-    public XWikiWikiReference(ResourceReference reference, String label, WikiParameters linkParameters,
+    private final XDOM labelXDOM;
+
+    /**
+     * Construct a new wiki reference.
+     *
+     * @param reference the reference the link points to
+     * @param label the already parsed label content
+     * @param linkParameters the parameters of the link
+     * @param freeStanding if the link is freestanding
+     */
+    public XWikiWikiReference(ResourceReference reference, XDOM label, WikiParameters linkParameters,
```

Comment:
Even if it's not public API it's still good to have a `@since` in new reusable stuff.

Reply:
Yes, I somehow forgot that, thank you. I've also added the `@since` to the new `getLabelXDOM` method.